app/services/warm_cache.rb

```diff
@@ -0,0 +1,36 @@
+class WarmCache
+  def call
+    r = {}
+    # Get all provider who have an api token that has been used in the past 6.months
```

Comment:
The comment might be outdated, you're query seems to get tokens that are 1 month old